pkg/fanal/image/oci.go

```diff
@@ -10,17 +10,21 @@ import (
 )
 
 func tryOCI(fileName string) (v1.Image, error) {
-	var inputTag, inputFileName string
+	var inputRef, inputFileName string
 
 	// Check if tag is specified in input
 	// e.g. /path/to/oci:0.0.1
-	if strings.Contains(fileName, ":") {
+	if strings.Contains(fileName, "@") {
+		splitFileName := strings.SplitN(fileName, "@", 2)
+		inputFileName = splitFileName[0]
+		inputRef = splitFileName[1]
+	} else if strings.Contains(fileName, ":") {
 		splitFileName := strings.SplitN(fileName, ":", 2)
```

Comment:
The code was written before Go 1.18. We can also rewrite it.

```suggestion
		inputFileName, inputRef := strings.Cut(fileName, ":")
```

Reply:
Yes, I have used the Cut function similarly to check for the tag/digest.